<!doctype html><html lang=en><meta charset=utf-8><meta name=generator content="Hugo 0.78.1"><meta name=viewport content="width=device-width,initial-scale=1,viewport-fit=cover"><meta name=color-scheme content="light dark"><meta name=supported-color-schemes content="light dark"><title>Nobile 2016&nbsp;&ndash;&nbsp;Papers</title><link rel=stylesheet href=/stevengogogo/PaperNotes/css/core.min.0e6154f1fe5441faaa4f418ca1a607b9f3edbad43b1ab8782737ee279ca2c5a48c8dbe23af3340af50c91a2558bc2b55.css integrity=sha384-DmFU8f5UQfqqT0GMoaYHufPtutQ7Grh4JzfuJ5yixaSMjb4jrzNAr1DJGiVYvCtV><meta name=twitter:card content="summary"><meta name=twitter:title content="Nobile 2016"><body><section id=header><div class="header wrap"><span class="header left-side"><a class="site home" href=/stevengogogo/PaperNotes/><span class="site name">Papers</span></a></span>
<span class="header right-side"><div class="nav wrap"><nav class=nav><a class="nav item" href=/stevengogogo/PaperNotes/categories/>Categories</a><a class="nav item" href=/stevengogogo/PaperNotes/tags/>Tags</a></nav></div></span></div><div class="site slogan"><span class=title>Collection of papers I read</span></div></section><section id=content><div class=article-container><section class="article header"><h1 class="article title">Nobile 2016</h1><p class="article date">2020-10-23</p></section><article class="article markdown-body"><p><a href=https://sciwheel.com/work/#/items/5237369 target=_blank>Sciwheel</a></p><p><a href=https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5862309/ target=_blank>Article</a><sup id=fnref:1><a href=#fn:1 class=footnote-ref role=doc-noteref>1</a></sup></p><h2 id=introduction>Introduction</h2><ul><li>Bioinformatics, Computational Biology and Systems Biology exploit either physico-chemical or mathematical modeling, characterized by different scales of granularity, abstraction levels and goals</li><li>All of them are computationally challenging</li><li>high-performance computing (HPC): computer clusters and grid computing, <strong>GPU</strong></li><li>From the point of view of the software developer, GPU programming still remains a challenging task (GPUs are quite different from CPUs)</li><li>Nvidia’s CUDA (Compute Unified Device Architecture) is the most used library</li><li>C/C++ are used most. Python is catching up.</li></ul><h2 id=sequence-alignment>Sequence alignment</h2><ul><li>high-throughput next-generation sequencing (NGS) methods</li><li>hash tables vs suffix/prefix trees</li></ul><h2 id=molecular-dynamics>Molecular dynamics</h2><ul><li>Molecular dynamics is computationally challenging: the length of the time step of a simulation is generally limited to &lt;5 fs, while the overall time of the phenomenon is, typically, in the order of ns or s.</li><li>A CUDA implementation of generalized explicit solvent all-atom classic molecular dynamics within the <strong>AMBER</strong> package (Titan V error warning!)</li></ul><h2 id=molecular-docking>Molecular docking</h2><ul><li>protein–protein, protein–ligand or protein–DNA complex</li><li>major challenge concerns the sampling of the conformational space</li></ul><h2 id=prediction-and-searching-of-molecular-structures>Prediction and searching of molecular structures</h2><h2 id=simulation-of-spatio-temporal-dynamics>Simulation of spatio-temporal dynamics</h2><h2 id=deterministic-simulation-ode>Deterministic simulation (ODE)</h2><ul><li>Parallel simulation of biological molecular networks</li><li>a CUDA implementation of the LSODA algorithm, named cuda-sim, was presented by Zhou et al.<ul><li>converting a SBML model into CUDA code</li></ul></li><li>Nobile et al.: cupSODA, cupSODA*L</li></ul><h2 id=stochastic-simulation-sdes-ssa>Stochastic simulation (SDEs, SSA)</h2><ul><li>randomness can be described either by means of Stochastic Differential Equations (SDEs) or using explicit mechanistic models e.g. stochastic simulation algorithm (SSA)</li><li>CUDA version of SSA: 130× speed-up with respect to the sequential simulation on the host computer.</li><li>Komarov and D’Souza: GPU-ODM</li><li>τ-leaping algorithm: <strong>cuTauLeaping</strong> by Nobile (three orders of magnitude faster)</li></ul><h2 id=spatial-simulation-pdes>Spatial simulation (PDEs)</h2><ul><li>Reaction-Diffusion (RD) models</li><li>Toolboxes for PDEs running on GPUs</li><li>Stochastic RD models: subvolumes (similiar to PDEs) vs agent-based models (ABMs, molecules are modeled as individual particles)</li></ul><h2 id=agent-based-models-abms>Agent-based models (ABMs)</h2><ul><li>FLAME is a general-purpose simulator of ABMs, which exploits GPU acceleration.</li><li><a href=http://www.flamegpu.com>www.flamegpu.com</a></li></ul><h2 id=applications-in-systems-biology>Applications in Systems Biology</h2><ul><li>sensitivity analysis, parameter estimation, parameter sweep analysis: large number of simulations to explore the high-dimensional search space of possible model parameterizations, suitable for GPUs</li><li>Nobile et al. proposed a parameter estimation methodology based on a multi-swarm version of Particle Swarm Optimization (PSO), using a CUDA-powered version of SSA</li><li>cuRE integrates this parameter estimation methodology with Cartesian Genetic Programming</li><li>ABC-SysBio, a <strong>Python-based</strong> and GPU-powered framework based on approximate Bayesian computation</li></ul><h2 id=discussion>Discussion</h2><ul><li>many of the most performing tools required a tailored implementation to fully leverage the GPU architecture and its theoretical peak performance<ul><li>skillful usage of shared memory and asynchronous data transfers</li><li>GPU-optimized data structures</li><li>naïve porting of an existing software to CUDA is generally doomed to failure.</li></ul></li><li>OpenCL, for instance, is an open standard suitable for parallel programming of heterogeneous systems. But is more difficult to code in OpenCL.</li><li>Swan: porting CUDA code to OpenCL without rewriting<ul><li>ROCm is doing the same thing</li></ul></li><li>A potential drawback of GPUs is the availability of <strong>memory</strong>, esp for genome-wide data<ul><li>NV link: links two or more GPUs</li></ul></li></ul><h2 id=references>References</h2><section class=footnotes role=doc-endnotes><hr><ol><li id=fn:1 role=doc-endnote><p>Nobile MS, Cazzaniga P, Tangherloni A, Besozzi D. Graphics processing units in bioinformatics, computational biology and systems biology. Brief Bioinform. 2016;18(5):870-885. <a href=https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5862309/ target=_blank>PMC5862309</a>. <a href=#fnref:1 class=footnote-backref role=doc-backlink>&#8617;&#xfe0e;</a></p></li></ol></section></article><section class="article author"><p class=name>Shao-Ting Chiu</p><div class=details><a class=item href=https://github.com/stevengogogo target=_blank><span class="iconfont icon-github"></span>&nbsp;stevengogogo</a></div></section></div><div class="article bottom"><section class="article navigation"><p><a class=link href=/stevengogogo/PaperNotes/posts/octavia-2012/><span class="iconfont icon-article"></span>Octavia 2012</a></p><p><a class=link href=/stevengogogo/PaperNotes/posts/nivala-2010/><span class="iconfont icon-article"></span>Nivala 2010</a></p></section></div></section><section id=footer><div class=footer-wrap><p class=copyright>Papers</p><p class=powerby><span>Powered&nbsp;by&nbsp;</span><a href=https://gohugo.io target=_blank>Hugo</a><span>&nbsp;&&nbsp;</span><a href=https://themes.gohugo.io/hugo-notepadium/ target=_blank>Notepadium</a></p></div></section><link rel=stylesheet href=https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.min.css integrity=sha384-zB1R0rpPzHqg7Kpt0Aljp8JPLqbXI3bhnPWROx27a9N0Ll6ZP/+DiW/UqRcLbRjq crossorigin=anonymous><script defer src=https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.min.js integrity=sha384-y23I5Q6l+B6vatafAwxRu/0oK/79VlbSz7Q9aiSZUvyWYIYsd+qj+o24G5ZU2zJz crossorigin=anonymous></script><script defer src=https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/contrib/auto-render.min.js integrity=sha384-kWPLUVMOks5AQFrykwIup5lo0m3iMkkHrD0uJ4H5cjeGihAutqP0yW0J6dpFiVkI crossorigin=anonymous onload=renderMathInElement(document.body);></script></body></html>